istioctl/cmd/sidecar.go

```diff
@@ -0,0 +1,291 @@
+
+	generateConfigCmd.PersistentFlags().StringVar(&gkeProject, "project", "", "Target project name")
+	generateConfigCmd.PersistentFlags().StringVar(&clusterLocation, "location", "", "Target cluster location")
+	generateConfigCmd.PersistentFlags().StringVar(&clusterName, "cluster", "", "Target cluster name")
+	return generateConfigCmd
+}
+
+func readWorkloadGroup(filename string, wg *clientv1alpha3.WorkloadGroup) error {
+	f, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+	if err = yaml.Unmarshal(f, wg); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Creates all the relevant config for the given workload group and cluster
+func createConfig(wg *clientv1alpha3.WorkloadGroup, cluster *containerpb.Cluster, outputName string) error {
+	temp, err := ioutil.TempDir("./", outputName)
+	if err != nil {
+		return err
+	}
+	// TODO: pack temp into a tarball and then delete folder
+	// defer os.RemoveAll(temp)
+
+	if err = createClusterEnv(wg, cluster, temp); err != nil {
+		return err
+	}
+	if err = createHosts(temp); err != nil {
+		return err
+	}
+	if err = createCertificates(temp); err != nil {
+		return err
+	}
+	if err = createMeshConfig(wg, temp); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Write cluster.env into the given directory
+func createClusterEnv(wg *clientv1alpha3.WorkloadGroup, cluster *containerpb.Cluster, dir string) error {
+	for _, v := range wg.Spec.Ports {
+		ports = append(ports, fmt.Sprint(v))
+	}
+	// default attributes and service name, namespace, ports, service account, service CIDR
+	clusterEnv := map[string]string{
+		"ISTIO_CP_AUTH":       "MUTUAL_TLS",
+		"ISTIO_INBOUND_PORTS": strings.Join(ports, ","),
+		"ISTIO_NAMESPACE":     wg.Namespace,
+		"ISTIO_PILOT_PORT":    "15012",
+		"ISTIO_SERVICE":       fmt.Sprintf("%s.%s", wg.Name, wg.Namespace),
+		"ISTIO_SERVICE_CIDR":  cluster.ServicesIpv4Cidr,
+		"SERVICE_ACCOUNT":     wg.Spec.ServiceAccount,
+	}
+	return ioutil.WriteFile(dir+"/cluster.env", []byte(mapToString(clusterEnv)), tempPerms)
+}
+
+// Create the needed hosts addition in the given directory
+func createHosts(dir string) error {
+	istiod, err := k8sService("istiod", "istio-system")
+	if err != nil {
+		return err
+	}
+	return ioutil.WriteFile(dir+"/hosts", []byte(fmt.Sprintf("%s\t%s", istiod.Spec.ClusterIP, "istiod.istio-system.svc")), tempPerms)
+}
+
+// Get and store the needed certificates
+// TODO: user internal generate-cert
+func createCertificates(dir string) error {
+	rootCert, err := k8sConfigMap("istio-ca-root-cert", "istio-system")
+	if err != nil {
+		return err
+	}
+	if err = ioutil.WriteFile(dir+"/root-cert.pem", []byte(rootCert.Data["root-cert.pem"]), tempPerms); err != nil {
+		return err
+	}
+
+	args := strings.Split("run istio.io/istio/security/tools/generate_cert -client -host spiffee://cluster.local/vm/vmname --out-priv key.pem --out-cert cert-chain.pem -mode citadel", " ")
```

Comment:
But if we do end up going with this approach, the `-host` flag here is not correct - should consist of NS and SA